### PR TITLE
Update overall structure of project

### PR DESCRIPTION
**Pull request by junha1**
This PR does many things. Sorry :(

What this PR does?
- Separate server and client
- Make network operations explicit: Only via `fetch()` and `broadcast()`
- Being a server is not the role of each module; it is enough to just run the associated DMS server

These are disabled because they are going to be changed a lot
- Some tests
- Some network-related methods

These are temporarily updated, and should be revisited soon.
- Node configurations
- Various node operations
- Fetch and broadcast implementations

### CODE REVIEW

**Review by JeongHunP — APPROVED** [review]

File: node/src/lib.rs [hunk 1]
```diff
@@ -63,6 +63,9 @@ pub struct Config {
     pub governance_port: u16,
     pub consensus_port: u16,
     pub repository_port: u16,
+
+    /// TODO: remove this and introduce a proper peer discovery protocol
```
Comment (TomTaehoonKim) [inline node/src/lib.rs:67]: Then will the `Node` structure hold the list of peers?

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

File: test-suite/src/lib.rs [hunk 2]
```diff
@@ -99,64 +99,62 @@ pub fn dispense_port() -> u16 {
...
         network_id: network_id.clone(),
-        ports: vec![(format!("dms-{network_id}"), dispense_port())]
-            .into_iter()
-            .collect(),
         members: Vec::new(),
         public_key,
         private_key,
+        ports: vec![(format!("dms-{network_id}"), dispense_port())]
+            .into_iter()
+            .collect(),
```
[the remark this answers is not in the document]
Comment (junha1) [reply test-suite/src/lib.rs:130]: What declaration?

[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply test-suite/src/lib.rs:130]: `ports` field is the second field in the declaration but here, it's in a different position.

[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply test-suite/src/lib.rs:130]: Oops my mistake

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by YoonhoKim0527 — APPROVED** [review]

**Pull request merged**